### PR TITLE
feature: isolate druid used in rm-datasource to solve druid compatibility issue

### PR DESCRIPTION
**Pull request by ggndnn**
### Ⅰ. Describe what this PR did
This PR was created to to solve druid compatibility issue, because druid was only used to do sql parser,  so all druid related code were isolated from users' druid by a separate class loader. All imports from druid in rm-datasource were removed except unit test code. In addition, sql parser part was extracted to several new projects as SPI, e.g., seata-sqlparser-core and seata-sqlparser-druid.

### CODE REVIEW

**Comment by xingfudeshi** [comment]
Good!

**Comment by ggndnn** [comment]
Wait #1703 merged.

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/DefaultDruidLoader.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        }
+        URL url = DefaultDruidLoader.class.getClassLoader().getResource(druidLocation);
+        if (url != null) {
+            try {
+                // extract druid.jar to temp file
+                // TODO use URLStreamHandler to handle nested jar loading in the future
+                File tempFile = File.createTempFile("seata", "sqlparser");
+                try (InputStream input = url.openStream()) {
+                    byte[] buf = new byte[1024];
+                    OutputStream output = new FileOutputStream(tempFile);
```
Comment (xingfudeshi) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/DefaultDruidLoader.java]: Keep this line within Try with resource statement too.

Comment (ggndnn) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/DefaultDruidLoader.java]: Done.

**Pull request merged**